pkg/controllers/user/serviceaccount/serviceaccount.go

```diff
@@ -0,0 +1,343 @@
+					return nil
+				}
+
+				// TODO what if we can't parse this?
+				split := strings.Split(annotation, ":")
+				clusterName, projectId := split[0], split[1]
+
+				projects, err := m.projects.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting projects: %v", err)
+				}
+
+				var project *v3.Project
+
+				for _, candidate := range projects {
+					if candidate.Name == projectId {
+						project = candidate
+					}
+				}
+
+				if project == nil {
+					return fmt.Errorf("no project with project id %v was found", projectId)
+				}
+
+				podSecurityPolicyTemplateId := project.Annotations[podSecurityPolicyTemplateLabel]
+
+				if podSecurityPolicyTemplateId == "" {
+					// check cluster
+					clusters, err := m.clusters.List("", labels.Everything())
+
+					if err != nil {
+						return fmt.Errorf("error getting clusters: %v", err)
+					}
+
+					var cluster *v3.Cluster
+
+					for _, candidate := range clusters {
+						if candidate.Name == clusterName {
+							cluster = candidate
+						}
+					}
+
+					if cluster == nil {
+						return fmt.Errorf("could not find cluster %v", clusterName)
+					}
+
+					podSecurityPolicyTemplateId = cluster.Annotations[podSecurityPolicyTemplateLabel]
+
+					if podSecurityPolicyTemplateId == "" {
+						logrus.Infof("START CLUSTER")
+						logJson(cluster)
+						logrus.Infof("END CLUSTER")
+						logrus.Infof("START PROJECT")
+						logJson(project)
+						logrus.Infof("END PROJECT")
+
+						logrus.Infof("No PSPTs found for project %v and cluster %v", projectId, clusterName)
+						return nil
+					}
+				}
+
+				templates, err := m.templates.List("", labels.Everything())
+
+				if err != nil {
+					return fmt.Errorf("error getting pod security policy templates: %v", err)
+				}
+
+				var template *v3.PodSecurityPolicyTemplate
+
+				for _, candidate := range templates {
+					if candidate.Name == podSecurityPolicyTemplateId {
+						template = candidate
+					}
+				}
+
+				if template == nil {
+					return fmt.Errorf("no template exists with id %v", podSecurityPolicyTemplateId)
+				}
+
+				template.Kind = "PodSecurityPolicy"
+				template.APIVersion = apiVersion
+				template.ResourceVersion = ""
+
+				objectMeta := template.ObjectMeta
+				objectMeta.Annotations[podSecurityAnnotation] = "true"
+
+				policy, err = m.policies.Create(&v1beta12.PodSecurityPolicy{
+					TypeMeta:   template.TypeMeta,
+					ObjectMeta: objectMeta,
+					Spec:       template.Spec,
+				})
+
+				if err != nil {
+					return fmt.Errorf("error creating pod security policy: %v", err)
+				}
+
+				logrus.Info("successfully created a pod security policy!")
+			}
+
+			roles, err := m.listRoles.List("", labels.Everything())
+
+			if err != nil {
+				return fmt.Errorf("error getting roles: %v", err)
+			}
+
+			var role *rbac.ClusterRole
+
+			for _, candidate := range roles {
+				if candidate.Annotations[podSecurityAnnotation] == "true" {
```

Comment:
As with PSPs, i dont think a value of true is sufficient. A cluster can have more than one clusterRole for managing different PSPTs across multiple projects. So, I think the value needs to be the UID of the specific PSPT it is for